zerver/views/invite.py

```diff
@@ -204,22 +202,22 @@ def resend_user_invite_email(
     return json_success(request)
 
 
+default_stream_ids: list[int] = []
```

Comment:
This doesn't seem like a good fix for the default issue; this created a singleton global list that every caller works with, in the same way that having it in the function definition does. Probably Pydantic has a way to specify a function (`return []`) that makes a new list as the default?

Reply:
Pydantic automatically handles[ mutable default values](https://docs.pydantic.dev/latest/concepts/models/#fields-with-non-hashable-default-values). 

It also has a `default_factory` parameter that can be used instead.

The convention for other python functions(not pydantic endpoints) is to assign `None` as default and then reassign in the function itself. This is what we have been doing for `Dicts` in the previous endpoints.

I will use the `default_factory` field to make sure it's clearer.